doc/stp/MSTP_HLD.md

```diff
@@ -0,0 +1,1060 @@
+
+<div align="center">
+<img src="images/MSTP_BPDU.png" alt="MSTP BPDU">
+<p>MSTP BPDU Format</p>
+</div><br>
+<div align="center">
+<img src="images/MSTI_config_message.PNG" alt="MSTP Config Message">
+<p>MSTI Configuration Messages</p>
+</div>
+
+*Refer to [RFC IEEE 802.1s-2002](https://standards.ieee.org/ieee/802.1s/1042/) for MSTP BPDU details.* 
+
+# Requirements
+1. Support the creation of Multiple Spanning Tree Instances (MSTIs).
+1. Support the assignment of one or more VLANs to a specific MSTI within a region.
+1. Support the option to assign a region name and revision number to MSTP regions in order to achieve unique identification of VLAN to instance mapping across switches.
+1. Support path selection and forwarding behaviour in MSTI to optimize network performance within each instance by configuring a distinct root bridge.
+1. Support the configuration of spanning tree parameters such as forward delay, hello timer, hop count and max age.
+1. The Destination Mac Address will be 01:80:C2:00:00:00 for MSTP BPDUs.
+1. Support compatibility with networks employing different spanning tree protocols, such as STP, RSTP and PVST via Protocol Migration. 
+1. Support edge port functionality
+1. Support BPDU guard functionality
+1. Support root guard functionality
+1. Support protocol operation on static breakout ports
+1. Support protocol operation on Port-channel interfaces
+
+# Architecture Design
+Following diagram explains the architectural design and linkages for MSTP. MSTP uses multiple existing SONiC containers, configuration details of each is mentioned below as well.
+
+![MSTP Architecture](images/MSTP_architecture.png)
+
+## STP Container
+STP Container is responsible for actions taken for BPDU rx and BPDU tx. Following are the details for implementation:
+
+### STPMgr
+Subscribes to CONFIG_DB and STATE_DB tables, parsing configurations and passes to STPd.
+
+### STPd
+Responsible for all MST protocol related calculations. BPDUs are sent and received in STPd and states are updated accordingly.
+
+### STPSync
+A process running as a part of STPD. Responsible for updating all the MSTP states in APP DB.
+
+The BPDU rx/tx, BPDU processing, handling of timers, handling of changes related to port or LAG using netlink events and STP port state sync to Linux Kernel will function the same as PVST.
+
+*Refer to [PVST's STP Container Details](https://github.com/sandeep-kulambi/SONiC/blob/631ab18211e7e396b138ace561b7a04e7f7b49a1/doc/stp/SONiC_PVST_HLD.md#34-stp-container)*
+
+## SWSS Container
+
+SWSS Container is responsible for passing on configurations to SAI as follows:
+
+### STPOrch
+Updates SAI via following APIs:
+
+1. Creating/deleting instances via SAI STP API.
+2. Assigning VLAN to instance via SAI STP API and SAI VLAN API.
+3. Creation of STP Port and assigning port state with respect to each instance via SAI STP API.
+4. Flushing FDB entries via SAI FDB API.
+
```

Comment:
BPDU guard functionality requires link-state control, maybe we need to add SAI api that control link state here as well.

Reply:
PortOrch already has link state control SAI attributes. STP notifies OrchAgent of any link state changes, which are then managed accordingly.